org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/views/IssueLocationsView.java

```diff
@@ -0,0 +1,402 @@
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Tree;
+import org.eclipse.ui.IEditorInput;
+import org.eclipse.ui.IEditorPart;
+import org.eclipse.ui.IEditorReference;
+import org.eclipse.ui.ISelectionListener;
+import org.eclipse.ui.ISharedImages;
+import org.eclipse.ui.IWorkbenchPage;
+import org.eclipse.ui.IWorkbenchPart;
+import org.eclipse.ui.IWorkbenchWindow;
+import org.eclipse.ui.PartInitException;
+import org.eclipse.ui.PlatformUI;
+import org.eclipse.ui.ide.IDE;
+import org.eclipse.ui.part.ViewPart;
+import org.eclipse.ui.texteditor.ITextEditor;
+import org.sonarlint.eclipse.core.SonarLintLogger;
+import org.sonarlint.eclipse.core.internal.SonarLintCorePlugin;
+import org.sonarlint.eclipse.core.internal.event.AnalysisEvent;
+import org.sonarlint.eclipse.core.internal.event.AnalysisListener;
+import org.sonarlint.eclipse.core.internal.markers.MarkerUtils;
+import org.sonarlint.eclipse.core.internal.markers.MarkerUtils.ExtraPosition;
+import org.sonarlint.eclipse.ui.internal.SonarLintImages;
+import org.sonarlint.eclipse.ui.internal.SonarLintUiPlugin;
+import org.sonarlint.eclipse.ui.internal.markers.ShowIssueFlowsMarkerResolver;
+
+/**
+ * Display details of a rule in a web browser
+ */
+public class IssueLocationsView extends ViewPart implements ISelectionListener, AnalysisListener {
+
+  public static final String ID = SonarLintUiPlugin.PLUGIN_ID + ".views.IssueLocationsView";
+
+  private LocationsViewer locationsViewer;
+  private IFile currentFile;
+  private ITextEditor currentEditor;
+  private IDocument currentDoc;
+
+  private static class FlowRootNode {
+
+    private final String label;
+    private final List<ExtraPosition> positions;
+
+    public FlowRootNode(String label, List<ExtraPosition> positions) {
+      this.label = label;
+      this.positions = positions;
+    }
+
+    public String getLabel() {
+      return label;
+    }
+
+    public List<ExtraPosition> getPositions() {
+      return positions;
+    }
+
+  }
+
+  private static class RootNode {
+
+    private final IMarker rootMarker;
+    private final List<List<ExtraPosition>> flows;
+
+    public RootNode(IMarker rootMarker, List<ExtraPosition> positions) {
+      this.rootMarker = rootMarker;
+      this.flows = rebuildFlows(positions);
+    }
+
+    private static List<List<ExtraPosition>> rebuildFlows(List<ExtraPosition> positions) {
+      List<List<ExtraPosition>> result = new ArrayList<>();
+      List<ExtraPosition> roots = positions.stream().filter(p -> p.getParent() == null).collect(Collectors.toList());
+      for (ExtraPosition root : roots) {
+        List<ExtraPosition> flow = new ArrayList<>();
+        Optional<ExtraPosition> current = Optional.of(root);
+        while (current.isPresent()) {
+          ExtraPosition currentValue = current.get();
+          flow.add(currentValue);
+          current = positions.stream().filter(p -> p.getParent() == currentValue).findFirst();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1698)